go/extra/ba/cmd/ba.go

```diff
@@ -0,0 +1,485 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common/metrics"
+	testCmd "github.com/oasislabs/oasis-core/go/oasis-test-runner/cmd"
+	testOasis "github.com/oasislabs/oasis-core/go/oasis-test-runner/oasis"
+)
+
+const (
+	cfgMetrics                = "metrics"
+	cfgMetricsP               = "m"
+	cfgMetricsTargetGitBranch = "metrics.target.git_branch"
+	cfgMetricsSourceGitBranch = "metrics.source.git_branch"
+)
+
+var (
+	cmpFlags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	cmpCmd = &cobra.Command{
+		Use:   "cmp",
+		Short: "compare last two benchmark instances",
+		Long: `cmp connects to prometheus, fetches the metrics of the benchmark instances and
+compares them. By default, the most recent instance (source) is fetched and
+compared to the pre-last (target). If --metrics.{target|source}.git_branch is
+provided, it compares the most recent instances in the corresponding branches.
+cmp compares all metrics provided by -m parameter and computes ratio source/target
```

Comment:
I think it's nicer if long parameter name is used in the help string: `-m parameter` -> `--metrics parameter`